clang/lib/AST/ASTContext.cpp

```diff
@@ -2272,8 +2272,8 @@ TypeInfo ASTContext::getTypeInfoImpl(const Type *T) const {
 #include "clang/Basic/AMDGPUTypes.def"
 #define HLSL_INTANGIBLE_TYPE(Name, Id, SingletonId) case BuiltinType::Id:
 #include "clang/Basic/HLSLIntangibleTypes.def"
-      Width = 0;
-      Align = 8;
+      Width = Target->getPointerWidth(LangAS::Default);
```

Comment:
Is this saying that an intangible type is the same size as a pointer type?  Is that right?

Reply:
Yes. Conceptually we wanted the intangible types to be sizeless, but since the codegen is translating them to llvm target type that has a size, the size of the source type must be non-zero as well.